src/MemStore.cc

```diff
@@ -25,7 +25,7 @@
 #include "tools.h"
 
 /// shared memory segment path to use for MemStore maps
-static const auto MapLabel = "cache_mem_map";
+static const auto MapLabel = "cache_mem";
 /// shared memory segment path to use for the free slices index
 static const char *SpaceLabel = "cache_mem_space";
```

Comment:
Since `cache_mem_space` presumably works while `cache_mem_map` does not, and neither is anywhere close to the limit, I assume we are exceeding 31 character limit when combining this MapLabel with other strings to form the final segment path. Suffixes like those added by StoreMapFileNosId() are pretty short too.

Please give at least one example of the offending final/combined segment name. We might want to shorten something else. There may even be a way to detect length violations at compile time, but it is probably too difficult.

Reply:
This is in cache.log on startup with ```workers 2```

```
2024/04/17 21:11:57| FATAL: Ipc::Mem::Segment::create failed to shm_open(/squid-cache_mem_map_filenos.shm): (63) File name too long
```
The segment name is 32 characters long.
From the man page for shm_open:
```
     [ENAMETOOLONG]     name exceeded the name size limit.  This is currently PSHMNAMLEN characters (defined in <sys/posix_shm.h>), but this may change in the future.
```

posic_shm.h contains
```c
#define PSHMNAMLEN      31      /* maximum name segment length we bother with */
```